src/router/call.js

```diff
@@ -13,54 +16,65 @@ var collapse = pathUtils.collapse;
  */
 module.exports = function routerCall(callPath, args,
                                      refPathsArg, thisPathsArg) {
-    var refPaths = normalizePathSets(refPathsArg || []);
-    var thisPaths = normalizePathSets(thisPathsArg || []);
-
-    var jsongCache = {};
     var router = this;
-    var action = runCallAction(router, callPath, args,
-                               refPaths, thisPaths, jsongCache);
-    var callPaths = [callPath];
-    return recurseMatchAndExecute(router._matcher, action, callPaths, call,
-                                  router, jsongCache).
 
-        // Take that
-        map(function(jsongResult) {
-            var reportedPaths = jsongResult.reportedPaths;
-            var jsongEnv = {
-                jsonGraph: jsongResult.jsonGraph
-            };
+    return Observable.defer(function() {
+
+        var refPaths = normalizePathSets(refPathsArg || []);
+        var thisPaths = normalizePathSets(thisPathsArg || []);
+        var jsongCache = {};
+        var action = runCallAction(router, callPath, args,
+                                   refPaths, thisPaths, jsongCache);
+        var callPaths = [callPath];
+
+        if (getPathsCount(refPaths) + 
```

Comment:
Same here. I imagine this is the exact same call impl as before just with the path count check, on subscription?